src/ORM/Association/BelongsToMany.php

```diff
@@ -1228,14 +1232,14 @@ function () use ($sourceEntity, $targetEntities, $primaryValue, $options) {
      * @param array $targetEntities entities in target table that are related to
      * the `$jointEntities`
      * @param array $options list of options accepted by `Table::delete()`
-     * @return array
+     * @return array|false Array of entities not deleted or false in case of deletion failure for atomic saves.
      */
     protected function _diffLinks(
         Query $existing,
         array $jointEntities,
         array $targetEntities,
         array $options = []
-    ): array {
+    ) {
```

Comment:
Why not ?array
then the null would indicate that and we can keep the typehint in place.

Reply:
In pretty much all ORM methods save failure is indicated by returning false. So I prefer being consistent here. Since it's an internal method the lack of type hint won't really matter, plus static analyzers would catch any type errors based on docblock types.